scripts/binary_to_array.py

```diff
@@ -0,0 +1,289 @@
+          [--filename_identifier=filename_c_identifier]
+
+By default, the output source file will be named the same as the input file,
+but with .cc as the extension; the output header file will be named the
+same as the input file but with .h as the extension.
+
+By default, the data will be in an array named $NAME_data and the size will
+be in a constant named $NAME_length, and the filename will be stored in
+$NAME_filename. In all these cases, $NAME is the input filename (sans path and
+extension) with runs of non-alphanumeric characters changed to underscores. The
+header guard will be generated from the output header filename in a similar way.
+
+By default, the data will be placed in the root namespace. If the data is placed
+in the root namespace, it will be declared as a C array (using extern "C" if
+compiled in C++ mode).
+
+The actual size of $NAME_data is $NAME_length + 1, where it contains an extra
+0x00 at the end. When data is actually text, $NAME_data can be used as a valid C
+string directly.
+"""
+
+from os import path
+from re import sub
+import argparse
+import logging
+
+arg_parser = argparse.ArgumentParser()
+
+arg_parser.add_argument("--input", type=str, help="Input file containing binary data to embed")
+arg_parser.add_argument("--output_source", type=str,
+                    help="Output source file, defining the array data.")
+arg_parser.add_argument("--output_header", type=str,
+                    help="Output header file, declaring the array data.")
+arg_parser.add_argument("--array", type=str, help="Identifier for the array.")
+arg_parser.add_argument("--array_size", type=str, help="Identifier for the array size.")
+arg_parser.add_argument("--filename", type=str, help="Override file name in code.")
+arg_parser.add_argument("--filename_identifier", type=str, help="Where to put the filename.")
+arg_parser.add_argument("--header_guard", type=str,
+                    help="Header guard to #define in the output header.")
+arg_parser.add_argument("--cpp_namespace", type=str,
+                    help="C++ namespace to use. If blank, will generate a C array.")
+
+# How many hex bytes to display in a line. Each "0x00, " takes 6 characters, so
+# a width of 12 lets us fit within 80 characters.
+WIDTH = 12
+
+
+class Error(Exception):
+  """Exception raised by methods in this module."""
+  pass
+
+
+def header(header_guard, namespaces, array_name, array_size_name, fileid):
+  """Return a C/C++ header for the given array.
+
+  Args:
+    header_guard: Name of the HEADER_GUARD to define.
+    namespaces: List of namespaces, outer to inner.
+    array_name: Name of the array.
+    array_size_name: Name of the array size constant.
+    fileid: Name of the identifier containing the file name.
+
+  Returns:
+    A list of strings containing the C/C++ header file, line-by-line.
+  """
+
+  data = []
+  data.extend([
+      "// Copyright 2016 Google Inc. All Rights Reserved.",
```

Comment:
18?

Reply:
If we're checking in the resulting file, probably?

@paulb777 what's your take on this? (The context here is that this script will generate a source file.)